### PR TITLE
Load env variables from .env files for non-conda

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #5061

### CODE REVIEW

File: src/client/datascience/kernel-launcher/kernelEnvVarsService.ts [hunk 1]
```diff
@@ -48,22 +48,23 @@ export class KernelEnvironmentVariablesService {
                 });
         }
 
-        if (interpreter?.envType !== EnvironmentType.Conda) {
```
Comment (DonJayamanne) [inline src/client/datascience/kernel-launcher/kernelEnvVarsService.ts:51]: This was the problem, when dealing with non-conda we'd not load the env variables.

File: src/client/common/variables/environmentVariablesProvider.ts [hunk 2]
```diff
@@ -116,7 +116,8 @@ export class EnvironmentVariablesProvider implements IEnvironmentVariablesProvid
     }
     private getWorkspaceFolderUri(resource?: Uri): Uri | undefined {
         if (!resource) {
-            return;
+            const workspaceFolders = this.workspaceService.workspaceFolders || [];
```
Comment (DonJayamanne) [inline src/client/common/variables/environmentVariablesProvider.ts:119]: If we have just one workspace folder opened, then use that to load the env variables.

File: src/client/datascience/kernel-launcher/kernelEnvVarsService.ts (hunk 1 above)
Comment (DonJayamanne) [reply src/client/datascience/kernel-launcher/kernelEnvVarsService.ts:51]: Could easily repro on linux as well.

File: src/client/common/process/environmentActivationService.ts [hunk 3]
```diff
@@ -225,6 +225,12 @@ export class EnvironmentActivationService implements IEnvironmentActivationServi
             // Remove it from cache (if it exists).
             void this.memento.update(key, undefined);
         }
+        if (env && customEnvVars) {
```
Comment (DonJayamanne) [inline src/client/common/process/environmentActivationService.ts:228]: In debt will refactor this, its a bigger change, but I think the way we create Python execution services is very complicated now and we've regressed in certain areas hence the need to fix this change in a number of places now.

File: src/test/datascience/notebook/executionService.vscode.test.ts [hunk 4]
```diff
@@ -167,6 +167,27 @@ suite('DataScience - VSCode Notebook - (Execution) (slow)', function () {
 
         await Promise.all([runCell(cell), waitForTextOutput(cell, '\tho\n\tho\n\tho\n', 0, true)]);
     });
+    test.only('Verify loading of env variables form .env file', async function () {
+        if (IS_REMOTE_NATIVE_TEST) {
```
Comment (DonJayamanne) [inline src/test/datascience/notebook/executionService.vscode.test.ts:171]: This test runs on Local + Raw & Local + Jupyter

**Review by rchiodo — APPROVED** [review]

**Comment by simeld** [comment]
I still have this issue when running python code, but not when debugging.
See
https://github.com/simeld/bug-vscode-dotenv
for a repo with reproducible bug.

**Comment by DonJayamanne** [comment]
@simeld  please file an issue against the python extension repo

**Pull request merged**